src/server.cpp

```diff
@@ -4128,25 +4642,32 @@ int prepareForShutdown(int flags) {
     /* Kill the saving child if there is a background saving in progress.
        We want to avoid race conditions, for instance our saving child may
        overwrite the synchronous saving did by SHUTDOWN. */
-    if (g_pserver->rdb_child_pid != -1) {
+    if (g_pserver->child_type == CHILD_TYPE_RDB) {
         serverLog(LL_WARNING,"There is a child saving an .rdb. Killing it!");
         /* Note that, in killRDBChild, we call rdbRemoveTempFile that will
```

Comment:
This comment was removed in 6.2.2, based on my reading it's outdated now but I could be wrong.